### PR TITLE
Package advi.2.0.0

### PR DESCRIPTION
**Pull request by diremy**
### `advi.2.0.0`
Active DVI Dune package!
Active-DVI is a DVI previewer and presenter written in OCaml with some eye
candy effects for presentation, support for interactive demonstrations, and
embedding of arbitrary applications within the presentation.

Active-DVI is well suited for use in combination with WhizzyTeX
(http://cristal.inria.fr/whizzytex/)

---
* Homepage: http://advi.inria.fr
* Source repo: git+https://github.com:diremy/advi.git
* Bug tracker: Didier.Remy@inria.fr

---
:camel: Pull-request generated by opam-publish v2.0.3

### CODE REVIEW

File: packages/advi/advi.2.0.0/opam
Comment (avsm) [inline packages/advi/advi.2.0.0/opam:33]:
```suggestion
  "conf-texlive"
]
```

needed to pull in kpsewhich

**Comment by avsm** [comment]
Many thanks for the contribution, @diremy!  For this to pass CI (and for users with the `opam depext` plugin), it would be useful to have the OS dependencies recorded. The build currently fails with:

```
#=== ERROR while compiling advi.2.0.0 =========================================#
# context              2.0.8 | linux/x86_64 | ocaml-base-compiler.4.11.2 | pinned(https://github.com/diremy/advi/archive/advi-2.0.0.tar.gz)
# path                 ~/.opam/4.11/.opam-switch/build/advi.2.0.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build make
# exit-code            2
# env-file             ~/.opam/log/advi-15-ecfa18.env
# output-file          ~/.opam/log/advi-15-ecfa18.out
### output ###
# dune build --root=. src/Makefile.config
#         bash src/configure.sh (exit 1)
# (cd _build/default/src && /bin/bash discover.sh configure.sh)
# No kpsewhich command found, check your setup!
# make: *** [Makefile:28: src/Makefile.config] Error 1
```

I left a suggestion in the PR to add the conf-texlive package which already has the external dependencies for this. If you commit that suggestion to your branch, the CI will retest.

File: packages/advi/advi.2.0.0/opam
Comment (diremy) [reply packages/advi/advi.2.0.0/opam:33]: Thanks. I did not know about this package.   

However, this package only checks that the command pdflatex exists, while I instead need to check that the command "kpsewhich" exists.  The former may imply the later, but I am not entirely sure: there may be installations of latex that do not install pdflatex (which did not exist when advi was first written), athough I would be surprised if it were the case today.  

Conversely, I am not sure that the presence of pdflatex ensures that of kpsewhich, although again I would be surprised if it were not the case. 

Another question is why isn't there a depext formular for macports and just one for homebrew? 

Actually, the defaut detection of os-family by opam seems biaised towards homebrew: I have a full installation with macports and an empty installation of homebrew (with zero homebrew package installed) and I had to explicitly tell opam that my os-distribution family is macports, otherwise it could not compile conf-libssl. 

> Many thanks for the contribution, @diremy! For this to pass CI (and for users with the `opam depext` plugin), it would be useful to have the OS dependencies recorded. 

> The build currently fails with:
> 
> ```
> #=== ERROR while compiling advi.2.0.0 =========================================#
> # context              2.0.8 | linux/x86_64 | ocaml-base-compiler.4.11.2 | pinned(https://github.com/diremy/advi/archive/advi-2.0.0.tar.gz)
> # path                 ~/.opam/4.11/.opam-switch/build/advi.2.0.0
> # command              ~/.opam/opam-init/hooks/sandbox.sh build make
> # exit-code            2
> # env-file             ~/.opam/log/advi-15-ecfa18.env
> # output-file          ~/.opam/log/advi-15-ecfa18.out
> ### output ###
> # dune build --root=. src/Makefile.config
> #         bash src/configure.sh (exit 1)
> # (cd _build/default/src && /bin/bash discover.sh configure.sh)
> # No kpsewhich command found, check your setup!
> # make: *** [Makefile:28: src/Makefile.config] Error 1
> ```
> 
> I left a suggestion in the PR to add the conf-texlive package which already has the external dependencies for this. If you commit that suggestion to your branch, the CI will retest.

How should I proceed? Should I fix this on advi git and then resubmit another independent opam-publish pull request or is there another better way? 

Didier

Comment (mseri) [reply packages/advi/advi.2.0.0/opam:33]: You can modify the metadata in the opam package of this PR without having to retag and resubmit. Opam uses the opam-repository information for the installation, not the file in the repository (unless you pin that).

In general any recent enough texlive installation should come with `kpsewhich`, so I think the suggestion is the best bet. I don’t know if there is a separate package that you can specify for it, but it seems unnecessarily complicated and can always be improved in the future if needed.

as per brew vs macports, usually people use one or the other so this has never come up. I think it may be worth opening an issue and see if we can find a better default. I think brew becomes preferred mostly because appears earlier alphabetically.

Comment (mseri) [reply packages/advi/advi.2.0.0/opam:34]: @diremy I have committed the suggested change, let’s see if it is enough to unblock the tests. If so please make the change also in your repository (no need to re tag or republish)

Comment (mseri) [reply packages/advi/advi.2.0.0/opam:34]: The archlinux failure is surprising since from documentation it seems that texlive should ship the command: https://wiki.archlinux.org/index.php/TeX_Live

File: packages/advi/advi.2.0.0/opam
Comment (mseri) [inline packages/advi/advi.2.0.0/opam:34]:
```suggestion
  "conf-which"
]
```
Looking more carefully arch failure seems to be related to the use of which in the script

File: packages/advi/advi.2.0.0/opam
Comment (mseri) [inline packages/advi/advi.2.0.0/opam:24]:
```suggestion
license: "LGPL-2.1-or-later"
```

File: packages/advi/advi.2.0.0/opam
Comment (mseri) [inline packages/advi/advi.2.0.0/opam:26]: The inria gitlab page does not seem reachable. 
```suggestion
doc: "http://advi.inria.fr/"
```

do you also have an accessible dev repository?

**Review by mseri — CHANGES_REQUESTED** [review]

File: packages/advi/advi.2.0.0/opam
Comment (mseri) [reply packages/advi/advi.2.0.0/opam:24]: Is this the correct license?

**Comment by mseri** [comment]
I think with the fixes above the package should work everywhere it can and it will be ready to merge. Please let me know if you agree with them

**Comment by diremy** [comment]
> I think with the fixes above the package should work everywhere it can and it will be ready to merge. Please let me know if you agree with them

Yes, I agree will all changes and I reproduced them in  the source rep. 

 - I also added the dev-repo field as suggested.
 - I've also improved and fixed some messages in the advi-latex-files post installation script.
 - I have retag the current version with the same advi-2.0.0 tag, so that 

         git archive -o advi-2.0.0.tar.gz advi-2.0.0

   will include all these small changes.  

Would you automatically retrieve the new version, or should I make another "opam publish" pull-request? 

Didier

**Comment by mseri** [comment]
Thanks! I think if you `opam publish` with the new archive, it should force push on this PR, updating also the checksums (otherwise the next CI run would fail). So, please do that!

**Comment by diremy** [comment]
> Thanks! I think if you `opam publish` with the new archive, it should force push on this PR, updating also the checksums (otherwise the next CI run would fail). So, please do that!

Done.

File: packages/advi/advi.2.0.0/opam
Comment (mseri) [inline packages/advi/advi.2.0.0/opam:37]:
```suggestion
    ["dune" "subst"] {dev}
```
Sorry, I had missed this

Comment (mseri) [reply packages/advi/advi.2.0.0/opam]: You don't need to re-tag or re-release for this after you make the change upstream.

File: packages/advi/advi.2.0.0/opam
Comment (mseri) [inline packages/advi/advi.2.0.0/opam:34]: This improved but still fails on archlinux due to missing ghostscript seems.
Probably we need to add  `conf-ghostscript` for that, @avsm do you have a suggestion?
```

#=== ERROR while compiling advi.2.0.0 =========================================#
# context              2.0.8 | linux/x86_64 | ocaml-base-compiler.4.12.0 | pinned(https://github.com/diremy/advi/archive/advi-2.0.0.tar.gz)
# path                 ~/.opam/4.12/.opam-switch/build/advi.2.0.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build make
# exit-code            2
# env-file             ~/.opam/log/advi-13-2bfeb4.env
# output-file          ~/.opam/log/advi-13-2bfeb4.out
### output ###
# dune build --root=. src/Makefile.config
#         bash src/configure.sh (exit 1)
# (cd _build/default/src && /usr/sbin/bash discover.sh configure.sh)
# which: no gs in (/home/opam/.opam/4.12/.opam-switch/build/advi.2.0.0/_build/install/default/bin:/home/opam/.opam/4.12/bin:/home/opam/.opam/plugins/bin:/home/opam/.opam/4.12/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin)
# warning: kpathsea: configuration file texmf.cnf not found in these directories: /usr/bin:/usr/bin/share/texmf-local/web2c:/usr/bin/share/texmf-dist/web2c:/usr/bin/share/texmf/web2c:/usr/bin/texmf-local/web2c:/usr/bin/texmf-dist/web2c:/usr/bin/texmf/web2c:/usr:/usr/share/texmf-local/web2c:/usr/share/texmf-dist/web2c:/usr/share/texmf/web2c:/usr/texmf-local/web2c:/usr/texmf-dist/web2c:/usr/texmf/web2c://texmf-local/web2c:/://share/texmf-local/web2c://share/texmf-dist/web2c://share/texmf/web2c://texmf-local/web2c://texmf-dist/web2c://texmf/web2c.
# kpsewhich command:
# arcticle.cls
# failed, check your setup!
# make: *** [Makefile:28: src/Makefile.config] Error 1
```

**Comment by diremy** [comment]
Done. Didier

File: packages/advi/advi.2.0.0/opam
Comment (kit-ty-kate) [inline packages/advi/advi.2.0.0/opam:28]: You were using the `ocaml-base-compiler` instead of the `ocaml` package. The latter includes ocaml installation in /usr/bin (via ocaml-system) and variants (flambda, …). As a general rule of thumb, only the `ocaml` package should be used.

File: packages/advi/advi.2.0.0/opam
Comment (kit-ty-kate) [inline packages/advi/advi.2.0.0/opam]: The previous syntax used does not work. The semicolon is usually used in conjunction with the ssh syntax (e.g. `git@github.com:diremy/advi.git`) which is only accessible to people with write access

**Comment by mseri** [comment]
Thanks a lot @kit-ty-kate! @diremy can you upstream the changes to the opam file?

**Comment by mseri** [comment]
Feel free to announce the release on the discuss.ocaml.org forum, we have a community category and an announcement tag for this purpose

**Comment by diremy** [comment]
On 14/04/2021 23:25, Marcello Seri wrote:
> Thanks a lot @kit-ty-kate <https://github.com/kit-ty-kate>! @diremy
> <https://github.com/diremy> can you upstream the changes to the opam file?

I already did. Thank you both for ypur help!   -Didier

**Pull request merged**
